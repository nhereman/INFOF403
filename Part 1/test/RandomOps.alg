co random testing operations co

begin
  a := (-3)
  a := (a*a)*(-a*-a)
  read(b)

  if not a = b and a = 0 or a >= b and b > 5 then

    a := b

  fi
  print(a)
end