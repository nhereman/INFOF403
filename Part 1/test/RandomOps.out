token: begin	lexical unit: BEG
token: a	lexical unit: VARNAME
token: :=	lexical unit: ASSIGN
token: (	lexical unit: LEFT_PARENTHESIS
token: -	lexical unit: MINUS
token: 3	lexical unit: NUMBER
token: )	lexical unit: RIGHT_PARENTHESIS
token: a	lexical unit: VARNAME
token: :=	lexical unit: ASSIGN
token: (	lexical unit: LEFT_PARENTHESIS
token: a	lexical unit: VARNAME
token: *	lexical unit: TIMES
token: a	lexical unit: VARNAME
token: )	lexical unit: RIGHT_PARENTHESIS
token: *	lexical unit: TIMES
token: (	lexical unit: LEFT_PARENTHESIS
token: -	lexical unit: MINUS
token: a	lexical unit: VARNAME
token: *	lexical unit: TIMES
token: -	lexical unit: MINUS
token: a	lexical unit: VARNAME
token: )	lexical unit: RIGHT_PARENTHESIS
token: read	lexical unit: READ
token: (	lexical unit: LEFT_PARENTHESIS
token: b	lexical unit: VARNAME
token: )	lexical unit: RIGHT_PARENTHESIS
token: if	lexical unit: IF
token: not	lexical unit: NOT
token: a	lexical unit: VARNAME
token: =	lexical unit: EQUAL
token: b	lexical unit: VARNAME
token: and	lexical unit: AND
token: a	lexical unit: VARNAME
token: =	lexical unit: EQUAL
token: 0	lexical unit: NUMBER
token: or	lexical unit: OR
token: a	lexical unit: VARNAME
token: >=	lexical unit: GREATER_EQUAL
token: b	lexical unit: VARNAME
token: and	lexical unit: AND
token: b	lexical unit: VARNAME
token: >	lexical unit: GREATER
token: 5	lexical unit: NUMBER
token: then	lexical unit: THEN
token: a	lexical unit: VARNAME
token: :=	lexical unit: ASSIGN
token: b	lexical unit: VARNAME
token: fi	lexical unit: FI
token: print	lexical unit: PRINT
token: (	lexical unit: LEFT_PARENTHESIS
token: a	lexical unit: VARNAME
token: )	lexical unit: RIGHT_PARENTHESIS
token: end	lexical unit: END
Identifiers
a 4
b 6
